core/src/com/unciv/logic/automation/UnitAutomation.kt

```diff
@@ -267,6 +310,96 @@ object UnitAutomation {
         unit.movement.headTowards(settlerOrGreatPersonToAccompany.currentTile)
         return true
     }
+    
+    private fun tryPatrolBoarders(unit: MapUnit): Boolean {
+        if (unit.type == UnitType.Scout) return false
+
+        val friendlyCitiesTiles = unit.civInfo.cities
+                .filter { unit.civInfo == it.civInfo }
+                .flatMap { it.tiles }
+                .asSequence().map { unit.currentTile.tileMap.get(it) }
+        
+        val weakestInfluenceTile = friendlyCitiesTiles
+                .filter { it.getInfluence() > 0 && (it in unit.civInfo.viewableTiles) }
+                .sortedByDescending { it.getInfluence() }
+                .firstOrNull { unit.movement.canReach(it) }
+        
+        if (weakestInfluenceTile != null) {
+            if (unit.type.isLandUnit() && weakestInfluenceTile.isWater) return false
```

Comment:
We should filter these tiles on line 323, *before* we sort by descending, if these units won't be able to help there anyway